Monsters hunting persistence/src/main/java/cz/fi/muni/pa165/dao/AreaDaoImpl.java

```diff
@@ -0,0 +1,71 @@
+package cz.fi.muni.pa165.dao;
+
+import cz.fi.muni.pa165.entity.Area;
+import cz.fi.muni.pa165.enums.AreaType;
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+import javax.persistence.PersistenceContext;
+import static org.assertj.core.util.Preconditions.checkNotNull;
+
+/**
+ *
+ * @author Jan Gol <gol.honza@gmail.com>
+ */
+public class AreaDaoImpl implements AreaDao {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Override
+    public void create(Area area) {
+        if (area == null) {
+            throw new IllegalArgumentException("Area cannot be null!");
+        }
+        entityManager.persist(area);
+    }
+
+    @Override
+    public void delete(Area area) {
+        if (area == null) {
+            throw new IllegalArgumentException("Area cannot be null!");
+        }
+        entityManager.remove(area);
+    }
+
+    @Override
+    public Area findById(Long id) {
+        if (id == null || id < 0) {
+            throw new IllegalArgumentException("Id cannot be null or negative!");
```

Comment:
Split this into two 'ifs'. First, check if it is null, then if it is a positive number. When debugging, we need to know exactly what happened, not that happend one of two options.